File: codewars/7kyu/shortest-word/dduck_shortest-word.py
# https://www.codewars.com/kata/shortest-word/

find_short = lambda s: len(sorted(s.split(), key=len)[0])

print(find_short("bitcoin take over the world maybe who knows perhaps"))
#
#
# const findShort = (s) => s.split(' ')
#   .sort((a, b) => a.length - b.length)[0].length;
#
# findShort("bitcoin take over the world maybe who knows perhaps");
# //?
# // 3

File: codewars/7kyu/shortest-word/dduck_shortest-word.rs
// https://www.codewars.com/kata/shortest-word/

fn main() {
    fn find_short(_names: &str) -> usize {
        let mut temp: Vec<&str> = _names.split(" ").collect();
        temp.sort_by(|a, b| Ord::cmp(&a.len(), &b.len()));
        return temp[0].len();
    }

    println!("{:?}", find_short("bitcoin take over the world maybe who knows perhaps"));
    // 3
}
